### DISCUSSION TITLE
v2.2 - small updates and fixes

### DISCUSSION THREAD

**Discussion by Itai-Nelken** [Itai-Nelken/aptpac#11] [Releases]
## Changelog
- add workflows for github-actions for daily builds for each commit and pr and for checking if compilation is succesfull (for C-edition).
- error and exit if no packages are provided in `show` and `show-all` commands (C-edition).
- add `-DLEARN=1` and `-DCALLCOMMAND="xxx"` to CMakeLists.txt to compile with learning mode on and the command to use as the program name when printing the help respectively (C-edition).
- If no option is passed for `CALLCOMMAND`, use default `aptpac` value, if a value that isn't `aptpac` or `./aptpac` is given, give a warning and use default `aptpac` as well.
- Improvement to about function (shown when running `aptpac --version`).

## Installation
Check the [readme](https://github.com/Itai-Nelken/aptpac#installation)

## Uninstallation
Check the [readme](https://github.com/Itai-Nelken/aptpac#uninstalling) as well.

<hr /><em>This discussion was created from the release <a href='https://github.com/Itai-Nelken/aptpac/releases/tag/2.2'>v2.2 - small updates and fixes</a>.</em>

**Comment by Itai-Nelken** [comment]
Working on being able to turn learning mode on and off from the program itself without having to recompile each time.
help is welcome!

**Reply by CleanMachine1** [reply]
Consider having ./aptpac as a file which only forwards to a different binary.

Then make both binaries (learning and norm), them in /opt/.

./aptpac might just be a sh script, and it defaults to norm, then if the user runs ./aptpac -l then use the learning mode

**Reply by Itai-Nelken** [reply]
I prefer to have a single executable. I'm working on saving the setting for the bash version, and once I have a method that works, I'll port it to the C version.
Thanks for the suggestion!
